### PR TITLE
[8.x] Model Broadcasting - Adding broadcastWith() and broadcastAs() support

### PR DESCRIPTION
**Pull request by clemblanco**
This PR adds two optional model methods `broadcastAs()` and `broadcastWith()` when using Model Broadcasting.

--- 

When using [broadcasting basic Events](https://laravel.com/docs/8.x/broadcasting#broadcasting-events), we can specify which [broadcast data](https://laravel.com/docs/8.x/broadcasting#broadcast-data) and [broadcast name](https://laravel.com/docs/8.x/broadcasting#broadcast-name) should be used using both `broadcastAs()` and `broadcastWith()`.

However, when using [Model Broadcasting](), we don't have that flexibility as it stands.

Similar to what the framework already does for `broadcastOn()`, we could use:

```php
public function broadcastAs($event)
{
    return match($event) {
        'created' => 'server.created',
    };
}
```

```php
public function broadcastWith($event)
{
    return match($event) {
        'created' => ['id' => $this->user->id],
    };
}
```

Just like `broadcastOn()`:

1. not defining these methods will default to Laravel's default behaviour
2. not specifying a default within your `match()` statement will default to Laravel's default behaviour
3. if you'd like to always see the same broadcast data and/or name published, no matter the event, you could omit the `match()` expression (or `switch()` for php 7) and always return the same string or array

**Bonus: missing unit tests**

- The ability to use `broadcastOn()` to specify a channel for given events was not covered
- The default behaviours for the broadcast data and name were not covered

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -81,7 +82,6 @@
         "aws/aws-sdk-php": "^3.155",
         "doctrine/dbal": "^2.6|^3.0",
         "filp/whoops": "^2.8",
-        "guzzlehttp/guzzle": "^6.5.5|^7.0.1",
```
Comment (driesvints) [inline composer.json:84]: Please revert these changes to the dependencies.

Comment (clemblanco) [reply composer.json]: Yup sorry my bad, just pushed the revert as you were commenting.

Comment (clemblanco) [reply composer.json]: I'll have a look at the failing tests too.

**Comment by driesvints** [comment]
Converted this to a draft until you've managed to fix the tests. Feel free to mark as ready when you're done.

**Comment by clemblanco** [comment]
@driesvints It's now ready even thought it seems like PHP 8.1 tests are marked as passed while it looks like GitHub Actions say otherwise.

image

Are you aware of this?

**Comment by driesvints** [comment]
@clemblanco yes, deel free to ignore those.

**Comment by clemblanco** [comment]
@driesvints In that case it's ready for you to have a look. Let me know what you think.✌️
